lib/request.ts

```diff
@@ -45,18 +76,26 @@ export const request = (options: ServiceClientRequestOptions): Promise<ServiceCl
   }
 
   const httpRequestFn = options.protocol === 'https:' ? httpsRequest : httpRequest
-  return new Promise((resolve, reject) => {
+  return new Promise((resolve, originalReject) => {
     let hasRequestEnded = false
     let startTime: [number, number]
     let timings: Timings
+    let reject = originalReject
     if (options.timing) {
       startTime = process.hrtime()
       timings = {
-        lookup: -1,
-        socket: -1,
-        connect: -1,
-        response: -1,
-        end: -1
+        lookup: undefined,
+        socket: undefined,
+        connect: undefined,
+        response: undefined,
+        end: undefined
+      }
+      reject = (error: Error) => {
+        Object.assign(error, {
```

Comment:
Now I noticed that `Promise` cannot have a custom error type as generic. So we cannot pass the `ErrorWithTimings` type from `request.ts` to `client.ts`. Now I'd like to leave this as is and guard these fields with unit tests.

Does anyone have suggestions?

Reply:
I believe it would still be beneficial to create `ErrorWithTimings` as a subclass of errors. It will still satisfy the `Error` type and in client code you can do `instanceof` check to get type safe access to fields

Reply:
Thanks for your advice. I’ll push commits tomorrow. My apartment’s internet doesn’t work today somehow...